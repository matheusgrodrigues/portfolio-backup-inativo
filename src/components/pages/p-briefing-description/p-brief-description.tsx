// Organism
import { OHeader } from "../../organisms/o-header";
import { OFooter } from "../../organisms/o-footer";

interface IPBriefDescription {}

export const PBriefDescription = ({}: IPBriefDescription) => {
  return (
    <>
      <OHeader
        avatar={{ url: "/images/avatar.jpeg", alt: "" }}
        toggleTheme={() => null}
      />

      <OFooter />
    </>
  );
};
