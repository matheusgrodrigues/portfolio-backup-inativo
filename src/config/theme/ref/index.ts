import { colors } from './color'
import { spacing } from './spacing'
import { font } from './font'
import { bp } from './breakpoint'

export { colors, spacing, font, bp }
