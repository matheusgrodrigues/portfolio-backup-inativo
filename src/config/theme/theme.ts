const fontFamily = {
    fontFamily_inter: 'Inter',
};

export type FontSizeName = 'fontSize_textSm' | 'fontSize_textMd' | 'fontSize_textXl' | 'fontSize_displayLg';

const fontSize = {
    fontSize_textSm: '14px',
    fontSize_textMd: '16px',
    fontSize_textXl: '20px',
    fontSize_displayLg: '48px',
};

export type FontWeightName = 'fontWeight_semibold' | 'fontWeight_regular' | 'fontWeight_medium';

const fontWeight = {
    fontWeight_regular: '400',
    fontWeight_medium: '500',
    fontWeight_semibold: '600',
};

export type ColorName =
    | 'color_gradient_primary600'
    | 'color_primary500'
    | 'color_primary600'
    | 'color_gray500'
    | 'color_gray900'
    | 'color_gray50'
    | 'color_white'
    | 'color_black'
    | 'color_transparent';

const colors = {
    color_gradient_primary600: `linear-gradient(90deg, #7f56d9 0%, #9e77ed 100%);`,
    color_primary500: '#9e77ed',
    color_primary600: '#7f56d9',
    color_gray500: '#667085',
    color_gray900: '#101828',
    color_gray50: '#f9fafb',
    color_white: '#ffffff',
    color_black: '#000000',
    color_transparent: 'transparent',
};

export type SpacingName =
    | 'spacing_4'
    | 'spacing_8'
    | 'spacing_12'
    | 'spacing_16'
    | 'spacing_20'
    | 'spacing_24'
    | 'spacing_32'
    | 'spacing_40'
    | 'spacing_48'
    | 'spacing_64'
    | 'spacing_80'
    | 'spacing_96'
    | 'spacing_128'
    | 'spacing_160'
    | 'spacing_192'
    | 'spacing_224'
    | 'spacing_256';

const spacing = {
    spacing_4: '4px',
    spacing_8: '8px',
    spacing_12: '12px',
    spacing_16: '16px',
    spacing_20: '20px',
    spacing_24: '24px',
    spacing_32: '32px',
    spacing_40: '40px',
    spacing_48: '48px',
    spacing_64: '64px',
    spacing_80: '80px',
    spacing_96: '96px',
    spacing_128: '128px',
    spacing_160: '160px',
    spacing_192: '192px',
    spacing_224: '224px',
    spacing_256: '256px',
};

type BreakpointName = 'breakpoint_sm' | 'breakpoint_md' | 'breakpoint_lg' | 'breakpoint_xl';

const breakpoint = {
    breakpoint_sm: '640px',
    breakpoint_md: '768px',
    breakpoint_lg: '1024px',
    breakpoint_xl: '1280px',
};

export const bpHelper = (bp: BreakpointName, content: string) =>
    `@media screen and (min-width: ${breakpoint[bp]}) {
    ${content}
  }`;

export type ThemeName = 'light' | 'dark';

type Theme = {
    name: ThemeName;
    ref: {
        breakpoint: typeof breakpoint;
        fontFamily: typeof fontFamily;
        fontWeight: typeof fontWeight;
        fontSize: typeof fontSize;
        spacing: typeof spacing;
        colors: typeof colors;
    };
};

declare module 'styled-components' {
    export interface DefaultTheme extends Theme {}
}

export const themeLight = {
    name: 'light',
    ref: { breakpoint, fontFamily, fontWeight, fontSize, spacing, colors },
};

export const darkTheme = {
    name: 'dark',
    ref: { breakpoint, fontFamily, fontWeight, fontSize, spacing, colors },
};
