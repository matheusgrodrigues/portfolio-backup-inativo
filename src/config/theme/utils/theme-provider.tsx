'use client'

import { ThemeProvider as StyledThemeProvider } from 'styled-components'

export const ThemeProvider = StyledThemeProvider
