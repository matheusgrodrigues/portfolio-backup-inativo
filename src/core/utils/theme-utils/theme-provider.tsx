'use client';

import { ThemeProvider as StyledThemeProvider } from 'styled-components';

const ThemeProvider = StyledThemeProvider;

export default ThemeProvider;
