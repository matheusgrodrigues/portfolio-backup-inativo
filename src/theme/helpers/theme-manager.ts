import { DefaultTheme } from "styled-components";

export const getRefColor = (theme: DefaultTheme, color: string) => {
  return theme.ref.por_ref_colors[`${color}`];
};
