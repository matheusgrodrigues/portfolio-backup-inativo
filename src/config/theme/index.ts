import { colors, colorsDark, font, spacing, bp } from './ref'

export const bpHelper = (bp: string, content: string) => {
    return `
  @media screen and (min-width: ${bp}) {
    ${content}
  }
  `
}

export const themeLight = {
    ref: { colors, font, spacing, bp },
}

export const darkTheme = {
    ...themeLight,
    ref: { ...themeLight.ref, colors: { ...colorsDark } },
}
