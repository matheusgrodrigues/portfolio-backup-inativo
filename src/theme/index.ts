import { por_ref_colors, por_ref_colorsDark } from "./ref";

const theme = {
  ref: { por_ref_colors },
  sys: {},
  comp: {},
};

export const darkTheme = {
  ...theme,
  ref: { ...theme.ref, por_ref_colors: { ...por_ref_colorsDark } },
};
