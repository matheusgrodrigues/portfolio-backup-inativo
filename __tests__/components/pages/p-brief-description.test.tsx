import renderer from "react-test-renderer";

// Theme
import { ThemeProvider } from "@/src/theme/helpers/theme-provider";
import { themeLight } from "@/src/theme";

// Pages
import { PBriefDescription } from "@/src/components/pages/p-briefing-description/p-brief-description";

describe("Deve renderizar o PBriefDescription corretamente", () => {
  it("Deve preservar a estrutura do componente", () => {
    const three = renderer
      .create(
        <ThemeProvider theme={themeLight}>
          <PBriefDescription />
        </ThemeProvider>
      )
      .toJSON();

    expect(three).toMatchSnapshot();
  });
});
