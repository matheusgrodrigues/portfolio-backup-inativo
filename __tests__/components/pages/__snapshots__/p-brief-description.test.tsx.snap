// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o PBriefDescription corretamente Deve preservar a estrutura do componente 1`] = `
[
  <div
    className="sc-eDLKEg dfSAHy"
  >
    <span
      alt=""
      className="sc-beyTiQ lPxIG"
      data-testid="a-avatar"
      src="/images/avatar.jpeg"
    >
      <img
        alt=""
        className="sc-guDMob AvYEu"
        data-testid="a-avatar-image"
        src="/images/avatar.jpeg"
      />
    </span>
    <button
      className="sc-hLQTFJ iIVlID"
    >
      <svg
        fill="none"
        height={32}
        onClick={[Function]}
        viewBox="0 0 15 15"
        width="200px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M7.5 0C7.77614 0 8 0.223858 8 0.5V2.5C8 2.77614 7.77614 3 7.5 3C7.22386 3 7 2.77614 7 2.5V0.5C7 0.223858 7.22386 0 7.5 0ZM2.1967 2.1967C2.39196 2.00144 2.70854 2.00144 2.90381 2.1967L4.31802 3.61091C4.51328 3.80617 4.51328 4.12276 4.31802 4.31802C4.12276 4.51328 3.80617 4.51328 3.61091 4.31802L2.1967 2.90381C2.00144 2.70854 2.00144 2.39196 2.1967 2.1967ZM0.5 7C0.223858 7 0 7.22386 0 7.5C0 7.77614 0.223858 8 0.5 8H2.5C2.77614 8 3 7.77614 3 7.5C3 7.22386 2.77614 7 2.5 7H0.5ZM2.1967 12.8033C2.00144 12.608 2.00144 12.2915 2.1967 12.0962L3.61091 10.682C3.80617 10.4867 4.12276 10.4867 4.31802 10.682C4.51328 10.8772 4.51328 11.1938 4.31802 11.3891L2.90381 12.8033C2.70854 12.9986 2.39196 12.9986 2.1967 12.8033ZM12.5 7C12.2239 7 12 7.22386 12 7.5C12 7.77614 12.2239 8 12.5 8H14.5C14.7761 8 15 7.77614 15 7.5C15 7.22386 14.7761 7 14.5 7H12.5ZM10.682 4.31802C10.4867 4.12276 10.4867 3.80617 10.682 3.61091L12.0962 2.1967C12.2915 2.00144 12.608 2.00144 12.8033 2.1967C12.9986 2.39196 12.9986 2.70854 12.8033 2.90381L11.3891 4.31802C11.1938 4.51328 10.8772 4.51328 10.682 4.31802ZM8 12.5C8 12.2239 7.77614 12 7.5 12C7.22386 12 7 12.2239 7 12.5V14.5C7 14.7761 7.22386 15 7.5 15C7.77614 15 8 14.7761 8 14.5V12.5ZM10.682 10.682C10.8772 10.4867 11.1938 10.4867 11.3891 10.682L12.8033 12.0962C12.9986 12.2915 12.9986 12.608 12.8033 12.8033C12.608 12.9986 12.2915 12.9986 12.0962 12.8033L10.682 11.3891C10.4867 11.1938 10.4867 10.8772 10.682 10.682ZM5.5 7.5C5.5 6.39543 6.39543 5.5 7.5 5.5C8.60457 5.5 9.5 6.39543 9.5 7.5C9.5 8.60457 8.60457 9.5 7.5 9.5C6.39543 9.5 5.5 8.60457 5.5 7.5ZM7.5 4.5C5.84315 4.5 4.5 5.84315 4.5 7.5C4.5 9.15685 5.84315 10.5 7.5 10.5C9.15685 10.5 10.5 9.15685 10.5 7.5C10.5 5.84315 9.15685 4.5 7.5 4.5Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>,
  <footer
    className="sc-jTQDJr"
  >
    <p>
      Â© 2023 matheusgomesdev.
    </p>
    <div>
      <button
        className="sc-hLQTFJ iIVlID"
      >
        <svg
          fill="none"
          height={32}
          viewBox="0 0 15 15"
          width={32}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M7.49933 0.25C3.49635 0.25 0.25 3.49593 0.25 7.50024C0.25 10.703 2.32715 13.4206 5.2081 14.3797C5.57084 14.446 5.70302 14.2222 5.70302 14.0299C5.70302 13.8576 5.69679 13.4019 5.69323 12.797C3.67661 13.235 3.25112 11.825 3.25112 11.825C2.92132 10.9874 2.44599 10.7644 2.44599 10.7644C1.78773 10.3149 2.49584 10.3238 2.49584 10.3238C3.22353 10.375 3.60629 11.0711 3.60629 11.0711C4.25298 12.1788 5.30335 11.8588 5.71638 11.6732C5.78225 11.205 5.96962 10.8854 6.17658 10.7043C4.56675 10.5209 2.87415 9.89918 2.87415 7.12104C2.87415 6.32925 3.15677 5.68257 3.62053 5.17563C3.54576 4.99226 3.29697 4.25521 3.69174 3.25691C3.69174 3.25691 4.30015 3.06196 5.68522 3.99973C6.26337 3.83906 6.8838 3.75895 7.50022 3.75583C8.1162 3.75895 8.73619 3.83906 9.31523 3.99973C10.6994 3.06196 11.3069 3.25691 11.3069 3.25691C11.7026 4.25521 11.4538 4.99226 11.3795 5.17563C11.8441 5.68257 12.1245 6.32925 12.1245 7.12104C12.1245 9.9063 10.4292 10.5192 8.81452 10.6985C9.07444 10.9224 9.30633 11.3648 9.30633 12.0413C9.30633 13.0102 9.29742 13.7922 9.29742 14.0299C9.29742 14.2239 9.42828 14.4496 9.79591 14.3788C12.6746 13.4179 14.75 10.7025 14.75 7.50024C14.75 3.49593 11.5036 0.25 7.49933 0.25Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </button>
      <button
        className="sc-hLQTFJ iIVlID"
      >
        <svg
          fill="none"
          height={32}
          viewBox="0 0 15 15"
          width={32}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M2 1C1.44772 1 1 1.44772 1 2V13C1 13.5523 1.44772 14 2 14H13C13.5523 14 14 13.5523 14 13V2C14 1.44772 13.5523 1 13 1H2ZM3.05 6H4.95V12H3.05V6ZM5.075 4.005C5.075 4.59871 4.59371 5.08 4 5.08C3.4063 5.08 2.925 4.59871 2.925 4.005C2.925 3.41129 3.4063 2.93 4 2.93C4.59371 2.93 5.075 3.41129 5.075 4.005ZM12 8.35713C12 6.55208 10.8334 5.85033 9.67449 5.85033C9.29502 5.83163 8.91721 5.91119 8.57874 6.08107C8.32172 6.21007 8.05265 6.50523 7.84516 7.01853H7.79179V6.00044H6V12.0047H7.90616V8.8112C7.8786 8.48413 7.98327 8.06142 8.19741 7.80987C8.41156 7.55832 8.71789 7.49825 8.95015 7.46774H9.02258C9.62874 7.46774 10.0786 7.84301 10.0786 8.78868V12.0047H11.9847L12 8.35713Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
  </footer>,
]
`;
